apps/console/src/features/applications/components/wizard/minimal-application-create-wizard.tsx

```diff
@@ -655,6 +672,82 @@ export const MinimalAppCreateWizard: FunctionComponent<MinimalApplicationCreateW
         document.getElementById("notification-div").scrollIntoView({ behavior: "smooth" });
     };
 
+    /**
+     * Checks whether the application name is valid.
+     * @param value - Application name as a form value.
+     * @param validation - The validation object.
+     */
+    const validateApplicationName = async (value: FormValue, validation: Validation) => {
+        const appName: string = value.toString().trim();
+
+        if (!isNameValid(appName)) {
+            validation.isValid = false;
+            validation.errorMessages.push(
+                t("console:develop.features.applications.forms." +
+                    "spaProtocolSettingsWizard.fields.name.validations.invalid", {
+                    appName: value.toString(),
+                    characterLimit: ApplicationManagementConstants.FORM_FIELD_CONSTRAINTS.APP_NAME_MAX_LENGTH
+                })
+            );
+
+            return;
+        }
+        if (isAppNameReserved(appName)){
+            validation.isValid = false;
+            validation.errorMessages.push(
+                t("console:develop.features.applications.forms.generalDetails.fields.name.validations.reserved", {
+                    appName: appName,
+                    characterLimit: ApplicationManagementConstants.FORM_FIELD_CONSTRAINTS.APP_NAME_MAX_LENGTH
+                })
+            );
+
+            return;
+        }
+        let response: ApplicationListInterface = null;
+
+        try {
+            response = await getApplicationList(null, null, "name eq " + value.toString());
+        } catch (error) {
+            if (error.response && error.response.data && error.response.data.description) {
+                dispatch(addAlert({
+                    description: error.response.data.description,
+                    level: AlertLevels.ERROR,
+                    message: t("console:develop.features.applications.notifications.fetchApplications.error.message")
+                }));
+
+                return;
+            }
+
+            dispatch(addAlert({
+                description: t("console:develop.features.applications.notifications." +
+                    "fetchApplications.genericError.description"),
+                level: AlertLevels.ERROR,
+                message: t("console:develop.features.applications.notifications." +
+                    "fetchApplications.genericError.message")
+            }));
+        }
+
+        if (response?.applications?.length > 0) {
+            validation.isValid = false;
+            validation.errorMessages.push(
+                t("console:develop.features.applications.forms.generalDetails.fields.name.validations.duplicate")
+            );
+        }
```

Comment:
Can't we have these statements within the try `{}` ?

Reply:
Fixed with 19af1fe13b344dbcdcf0c978195222bdd38d10dc